plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckAnimator.kt

```diff
@@ -69,13 +72,13 @@ internal abstract class PuckAnimator<T>(
     locationRenderer = renderer
   }
 
-  fun animate(vararg targets: T, options: (ValueAnimator.() -> Unit)? = null) {
-    if (isRunning) {
-      cancel()
-    }
-    if (userConfiguredAnimator.isRunning) {
-      userConfiguredAnimator.cancel()
-    }
+  fun animate(
+    vararg targets: T,
+    onUpdate: ((T) -> Unit)? = null,
+    options: (ValueAnimator.() -> Unit)? = null
+  ) {
+    cancelRunning()
+    puckUpdateListener = onUpdate
```

Comment:
seems we only have one puckUpdateListener at a time, and it will be overwritten every time the new animate call is invoked. How about exposing the onPuckUpdatedListener directly from the PuckAnimator? so that the LocationPuckManager can just register `onPuckUpdatedListener` once and use the same listener for all the updates.

Reply:
Sounds reasonable. 
My idea was here that we do not need it in each but it seems we do.